src/Symfony/Component/Routing/Exception/MissingMandatoryParametersException.php

```diff
@@ -19,4 +19,41 @@
  */
 class MissingMandatoryParametersException extends \InvalidArgumentException implements ExceptionInterface
 {
+    private array $missingParameters = [];
+    private string $routeName = '';
+
+    /**
+     * @param array  $missingParameters
+     * @param string $routeName
+     * @param int    $code
+     */
+    public function __construct(/*array*/ $missingParameters = [], /*string*/ $routeName = '', /*int*/ $code = 0, \Throwable $previous = null)
```

Comment:
please remove the comments now :)
the default value should not be an array, because we want to make the argument required, isn't it?
I'd suggest putting the route name first and to detect the new signature based on the second argument (which shouldn't default to an array to trigger a deprecation by default.

Reply:
Gottcha ! I now have a *string* `$routeName` as first *required* argument, and the second ~optional~ argument `$missingParameters` defaults to `null` and is (still) used to detect the new signature.
Please confirm this finally complies ^^